newton/_src/utils/schema_resolver.py

```diff
@@ -0,0 +1,479 @@
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, ClassVar
+
+
+@dataclass
+class AttrSpec:
+    """
+    Specifies a USD attribute and its transformation function.
+
+    Args:
+        usd_name (str): The name of the USD attribute.
+        transform (Callable[[Any], Any] | None): A function to transform the attribute value.
+    """
+
+    usd_name: str
+    transform: Callable[[Any], Any] | None = None
+
+
+class EngineSchemaPlugin:
+    name: str
+    # prim_type -> variable -> list[AttrSpec]
+    mapping: dict[str, dict[str, list[AttrSpec]]]
+    extra_attr_prefixes: ClassVar[list[str]] = []
```

Comment:
In USD terms these are called property namespaces, we might want to align to that here.